portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,48 +14,112 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.*;
+import com.google.sps.data.CommentBlock;
+import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns comment data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private CommentBlock comments = new CommentBlock();
+
+    // If no name is provided, name is set to default
+    private final String defaultName = "Anonymous";
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        String numComString = request.getParameter("numComments");
+
+        int numComments;
+        try {
+            numComments = Integer.parseInt(numComString);
+
+        } catch (NumberFormatException e) {
+            numComments = -1;
+        }
+        int numComLogged = 0;
+
+        for (Entity entity : results.asIterable()) {
+            if(numComments == 0 || numComments > numComLogged){
+                String comment = (String) entity.getProperty("comment");
```

Comment:
can comment or name be null? what will happen?

It is not a good idea to convert this way. Maybe String.valueOf